### PR TITLE
Adds ensureWidgetIsVisible() function to scrollable.dart

### PR DESCRIPTION
**Pull request by HansMuller**
Adds ensureWidgetIsVisible() function to scrollable.dart

Set the scrollOffset of a widget's Scrollable ancestor so that the
widget is centered within the scrollable. A future CL will add
support for specifying exactly where the widget appears. The scroll
can be animated by specifying the animation: parameter.

Changed the duration Scrollable.scrollTo() parameter from a Duration
to an AnimationPerformance so that one can configure all aspects of
the animation. The caller may also listen to the animation to schedule
other work while it updates or when its status changes.

### CODE REVIEW

File: sky/packages/sky/lib/widgets/scrollable.dart [hunk 1]
```diff
@@ -220,6 +233,42 @@ Scrollable findScrollableAncestor({ Widget target }) {
...
+      : new Point(targetSize.width / 2.0, 0.0)
+  );
+
+  Size scrollableSize = (scrollable.root as RenderBox).size;
+  Point scrollableCenter = scrollable.localToGlobal(
+    scrollable.scrollDirection == ScrollDirection.vertical
+      ? new Point(0.0, scrollableSize.height / 2.0)
+      : new Point(scrollableSize.width / 2.0, 0.0)
+  );
+  double scrollOffsetDelta = targetCenter.y - scrollableCenter.y;
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/scrollable.dart]: scrollDirection ?

File: sky/packages/sky/lib/widgets/scrollable.dart [hunk 2]
```diff
@@ -220,6 +233,42 @@ Scrollable findScrollableAncestor({ Widget target }) {
...
+  BoundedBehavior scrollBehavior = scrollable.scrollBehavior;
+  double scrollOffset = (scrollable.scrollOffset + scrollOffsetDelta)
+    .clamp(scrollBehavior.minScrollOffset, scrollBehavior.maxScrollOffset);
+
+  if (scrollOffset != scrollable.scrollOffset) {
+    scrollable.setState(() {
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/scrollable.dart]: This shouldn't be necessary.

**Comment by abarth** [comment]
LGTM

File: sky/packages/sky/lib/widgets/scrollable.dart (hunk 1 above)
Comment (HansMuller) [inline sky/packages/sky/lib/widgets/scrollable.dart]: Oops, yes, I forgot to carry scrollDirection through here.

File: sky/packages/sky/lib/widgets/scrollable.dart (hunk 2 above)
Comment (HansMuller) [inline sky/packages/sky/lib/widgets/scrollable.dart]: You're right. It's out.

**Pull request merged**